capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+
+
+class TcpEstablishedWithNonTcpError(Error):
+    pass
+
+
+class AristaTpDuplicateTermError(Error):
+    pass
+
+
+class AristaTpFragmentInV6Error(Error):
+    pass
+
+
+class Config(object):
+    """config allows a configuration to be assembled easily.
+
+    when appending to the configuration object, the element should be indented
+    according to the arista traffic-policy style.
+
+    a text representation of the config can be extracted with str().
+
+    attributes:
+    indent: The number of leading spaces on the current line.
+    lines: the text lines of the configuration.
+
+    """
+
+    def __init__(self):
+        self.lines = []
+
+    def __str__(self):
+        return "\n".join(self.lines)
+
+    def Append(self, line_indent, line, verbatim=False):
+        """append one line to the configuration.
+
+        args:
+        - line_indent: config specific spaces prepended to the line
+        - line: the configuratoin string to append to the config.
+        - verbatim: append line without adjusting indentation. Default False.
+        """
+        if verbatim:
+            self.lines.append(line)
+            return
+
+        self.lines.append(line_indent + line.strip())
+
+
+class Term(aclgenerator.Term):
+    """represents an individual AristaTrafficPolicy term.
+
+    useful for the __str__() method. wher literally, everything intersting
+    happens.
+
+    attributes:
+     term: the term object from the policy.
+     term_type: string indicating type of term, inet, inet6 etc.
+     noverbose: boolean to disable verbosity.
+
+    """
+
+    _PLATFORM = "arista_tp"
+    _ACTIONS = {
+        "accept": "",
+        "deny": "drop",
+        "reject": "drop",               # unsupported action, convert to drop
+        "reject-with-tcp-rst": "drop",  # ibid
+        # "next": "continue",
+    }
+    AF_MAP = {
+        "inet": 4,
+        "inet6": 6,
+    }
+    # the following lookup table is used to map between the various types of
+    # filters the generator can render.  as new differences are
+    # encountered, they should be added to this table.  Accessing members
+    # of this table looks like:
+    #  self._TERM_TYPE('inet').get('saddr') -> 'source-address'
+    #
+    # it's critical that the members of each filter type be the same, that is
+    # to say that if _TERM_TYPE.get('inet').get('foo') returns something,
+    # _TERM_TYPE.get('inet6').get('foo') must return the inet6 equivalent.
+    _TERM_TYPE = {
+        "inet": {
+            "addr_fam": "ipv4",
+            "addr": "address",
+            "saddr": "source prefix",
+            "daddr": "destination prefix",
+            "protocol": "protocol",
+            "established": "established",
+            # tcp flags
+            "ack": "ack",
+            "fin": "fin",
+            "initial": "initial",
+            "psh": "syn",
+            "rst": "rst",
+            "syn": "syn",
+            "tcp-est": "established",
+            "urg": "urg",
```

Comment:
All the TCP flags seem to be the same between inet and inet6, is there a reason you added them here?